cmd/argo/commands/delete.go

```diff
@@ -52,7 +55,7 @@ func NewDeleteCommand() *cobra.Command {
 					ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: flags.namespace},
 				})
 			}
-			if all || flags.completed || flags.resubmitted || flags.prefix != "" || flags.labels != "" || flags.fields != "" || flags.finishedAfter != "" {
+			if hasFilterFlag {
```

Comment:
`--all-namespaces` was previously not in this conditional.
from a quick glance, not sure if that actually makes a difference though

Reply:
When `--all-namespaces` is _not_ set, the [client's namespace is set as a flag](https://github.com/argoproj/argo-workflows/blob/2229fbac5ed1c61beda4e313f4b8a15655d919a3/cmd/argo/commands/delete.go#L51) (a few lines above).

However, if it is set, I _believe_ a `list` would be necessary anyway, so I _think_, functionally, these are equivalent